packages/contracts-ops/contracts/test/RebootEthereumBridgeRouter.t.sol

```diff
@@ -0,0 +1,90 @@
+        setUpReboot(1, "bridgerouter");
+        // load proxies
+        tokenRegistry = TokenRegistryHarness(
+            address(getTokenRegistry(ethereum))
+        );
+        vm.label(address(tokenRegistry), "tokenRegistry");
+        bridgeRouter = IBridgeRouterHarness(address(getBridgeRouter(ethereum)));
+        vm.label(address(bridgeRouter), "bridgeRouter");
+        // upgrade to harness
+        setUp_upgradeTokenRegistryHarness();
+        setUp_upgradeBridgeRouterHarness();
+        // load necessary contracts
+        xAppConnectionManager = getXAppConnectionManager(ethereum);
+        vm.label(address(xAppConnectionManager), "XAppConnectionManager");
+        upgradeBeaconController = getUpgradeBeaconController(ethereum);
+        vm.label(address(upgradeBeaconController), "upgradeBeaconController");
+        tokenBeacon = UpgradeBeacon(payable(tokenRegistry.tokenBeacon()));
+        vm.label(address(tokenBeacon), "tokenBeacon");
+        bridgeToken = BridgeToken(beaconImplementation(tokenBeacon));
+        vm.label(address(bridgeToken), "bridgeToken");
+        // home needed for vm.expectCall
+        home = address(getHome("ethereum"));
+        vm.label(home, "home");
+        BridgeRouterBaseTest.setUp_testFixtures();
+    }
+
+    function setUp_upgradeTokenRegistryHarness() public {
+        tokenRegistryHarnessImpl = address(new TokenRegistryHarness());
+        vm.label(tokenRegistryHarnessImpl, "tokenRegistryHarnessImpl");
+        vm.writeJson(
+            vm.toString(tokenRegistryHarnessImpl),
+            outputPath,
+            bridgeAttributePath(ethereum, "tokenRegistry.implementation")
+        );
+        reloadConfig();
+        pushSingleUpgrade(tokenRegistryUpgrade(ethereum), ethereum);
+        prankExecuteRecoveryManager(
+            address(getGovernanceRouter(ethereum)),
+            getDomainNumber(ethereum)
+        );
+    }
+
+    function setUp_upgradeBridgeRouterHarness() public {
+        bridgeRouterHarnessImpl = address(new EthereumBridgeRouterHarness(address(getAccountant(ethereum))));
+        vm.label(bridgeRouterHarnessImpl, "bridgeRouterHarnessImpl");
+        vm.writeJson(
+            vm.toString(bridgeRouterHarnessImpl),
+            outputPath,
+            bridgeAttributePath(ethereum, "bridgeRouter.implementation")
+        );
+        reloadConfig();
+        pushSingleUpgrade(bridgeRouterUpgrade(ethereum), ethereum);
+        prankExecuteRecoveryManager(
+            address(getGovernanceRouter(ethereum)),
+            getDomainNumber(ethereum)
+        );
+    }
+
+    function test_setUp_rebootBridgeRouter() public {
+        // check that the harnesses have harness methods available
+        assertEq(bridgeRouterHarnessImpl, bridgeRouterUpgrade(ethereum).implementation);
+        assertEq(tokenRegistryHarnessImpl, tokenRegistryUpgrade(ethereum).implementation);
+        bridgeRouter.exposed_dust(address(this));
```

Comment:
are there any ethereum bridge router-specific harness fns that could be swapped in here?

Reply:
no, the harnesses are identical.